x/upgrade/keeper/keeper.go

```diff
@@ -265,6 +271,98 @@ func (k Keeper) GetUpgradeInfoPath() (string, error) {
 	return filepath.Join(upgradeInfoFileDir, UpgradeInfoFileName), nil
 }
 
+// CreateUpgradeProofInfo creates the proof info struct with all necessary information
+// for relayers to upgrade counterparty's IBC clients.
+func (k Keeper) CreateUpgradeProofInfo(ctx sdk.Context, height int64) (*types.ProofInfo, error) {
```

Comment:
nit, can we split the components of this PR into private funcs?

* Get client state and pack (any)
* get client state proof and check err
* get cons state and pack (any)
* get cons state proof and check err